### PR TITLE
Support changed files for Gitea PRs

### PR DESCRIPTION
**Pull request by qwerty287**
#1138

TODOs:
 - [x] add tests (just use existing and extend) for this testcase
 - [x] ignore error if gitea version is to low
 - [x] change docs acordingly

### CODE REVIEW

File: server/remote/gitea/gitea.go [hunk 1]
```diff
@@ -540,3 +557,39 @@ func getStatus(status model.StatusValue) gitea.StatusState {
...
+	user, err := _store.GetUser(repo.UserID)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := c.newClientToken(ctx, user.Token)
+	if err != nil {
+		return nil, err
+	}
+	return common.Paginate(func(page int) ([]string, error) {
```
Comment (6543) [inline server/remote/gitea/gitea.go:582]: check if the error is "to low gitea version" and return no errror an no changed files -> as we have it now

**Comment by 6543** [comment]
TODOs:
 - [ ] add tests (just use existing and extend) for this testcase
 - [ ] ignore error if gitea version is to low
 - [ ] change docs acordingly

File: server/remote/gitea/gitea.go (hunk 1 above)
Comment (6543) [reply server/remote/gitea/gitea.go:582]: asap we have init the client we can do ... but in terms of api cals there wont be any difference

Comment (6543) [reply server/remote/gitea/gitea.go:582]: (the sdk do check on it's own before calling the endpoint)

Comment (6543) [reply server/remote/gitea/gitea.go:588]: It's done on client init if you dont tell to do otherwhise

File: docs/docs/20-usage/20-pipeline-syntax.md [hunk 2]
```diff
@@ -412,8 +412,7 @@ when:
 
 :::info
 Path conditions are applied only to **push** and **pull_request** events.
-It is currently **only available** for GitHub, GitLab.
-Gitea only supports **push** at the moment ([go-gitea/gitea#18228](https://github.com/go-gitea/gitea/pull/18228)).
+It is currently **only available** for GitHub, GitLab and Gitea.
```
Comment (6543) [inline docs/docs/20-usage/20-pipeline-syntax.md:415]: could you clarify that 'sed s/Gitea/Gitea[1]/g' ...

[1] all gitea version >= 1.18.0 ...

File: server/remote/gitea/gitea.go [hunk 3]
```diff
@@ -540,3 +557,45 @@ func getStatus(status model.StatusValue) gitea.StatusState {
...
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := c.newClientToken(ctx, user.Token)
+	if err != nil {
+		return nil, err
+	}
+
+	if strings.Contains(client.CheckServerVersionConstraint("1.18.0").Error(), "does not satisfy version constraint") {
```
Comment (6543) [inline server/remote/gitea/gitea.go:583]:
```suggestion
	if client.CheckServerVersionConstraint("1.18.0") != nil {
```

File: server/remote/gitea/gitea.go [hunk 4]
```diff
@@ -540,3 +557,45 @@ func getStatus(status model.StatusValue) gitea.StatusState {
...
+		// version too low
+		return []string{}, nil
```
Comment (6543) [inline server/remote/gitea/gitea.go:585]:
```suggestion
		log.Debug ....
		return []string{}, nil
```

**Comment by 6543** [comment]
thanks :heart: now only tests are missing

I think you can just look into github remote as there already should be some in that you can copy/use as template

File: server/remote/gitea/gitea.go [hunk 5]
```diff
@@ -33,6 +33,7 @@ import (
 
 	"code.gitea.io/sdk/gitea"
 	"github.com/rs/zerolog/log"
+	"github.com/woodpecker-ci/woodpecker/server/store"
```
Comment (lafriks) [inline server/remote/gitea/gitea.go:36]: this should be moved to import group below

File: docs/docs/20-usage/20-pipeline-syntax.md [hunk 6]
```diff
@@ -412,8 +412,7 @@ when:
 
 :::info
 Path conditions are applied only to **push** and **pull_request** events.
-It is currently **only available** for GitHub, GitLab.
-Gitea only supports **push** at the moment ([go-gitea/gitea#18228](https://github.com/go-gitea/gitea/pull/18228)).
+It is currently **only available** for GitHub, GitLab and Gitea (versions newer than 1.18.0)
```
Comment (lafriks) [inline docs/docs/20-usage/20-pipeline-syntax.md:415]:
```suggestion
It is currently **only available** for GitHub, GitLab and Gitea (version 1.18.0 and newer)
```

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
added tests by using the mockup store

**Review by lafriks — APPROVED** [review]

**Comment by 6543** [comment]
We cant as we use gin framework

We would have to patch that upstream

**Pull request merged**